♻️ Merge PR: Refactor project structure for backend and frontend

juliaogris opened the pull request:
- Move all data aggregation `go` code to `./backend`
- Create `frontend` with `create-react-app`
- Update `frontend` to integrate with Makefile and Google Cloudbuilds
- Rework README.md

juliaogris commented on frontend/yarn.lock:
```diff
@@ -0,0 +1,10021 @@
+# THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
```
Apparently you real are supposed to commit these 10K lines:
https://yarnpkg.com/blog/2016/11/24/lockfiles-for-all/

camh-anz replied:
yeah. that's how you get reproducible builds. it's also handy when you add/update a dependency to see the diff to this file to see everything else that changes. I'm fine with this being checked in (but not that it is 10,000 lines. i mean really? how many leftpad's do we need?)

camh-anz commented on frontend/Makefile:
```diff
@@ -0,0 +1,22 @@
+	yarn start
+install: node_modules
+node_modules: package.json
+	yarn
```
add the command `touch $@` after yarn to ensure that the `node_modules` timestamp is updated. It is possible that yarn installs something in a subdirectory of `node_modules` and the top-level directory does not get an updated timestamp. then it is always out-of-date with respect to `packages.json` and it will keep running `yarn`.

`$@` is the make special variable for the current target.

juliaogris replied:
Thanks for the detailed explanation.
Done.

camh-anz commented on cloudbuild.yaml:
```diff
@@ -20,7 +16,7 @@ steps:
     name: 'gcr.io/cloud-builders/npm:current'
     entrypoint: 'make'
-    args: ['-C', 'frontend', 'ensure-coverage']
+    args: ['-C', 'frontend', 'check-coverage']
```
these changes to cloudbuild.yaml should be to add all the front-end steps, since until this commit there is no frontend Makefile so cloudbuild is broken.

camh-anz commented on frontend/Makefile:
```diff
@@ -0,0 +1,22 @@
+all: check-coverage lint build
+check-coverage:  test
```
nit: two spaces between colon and `test`

juliaogris replied:
Done.

juliaogris commented:
> The second commit (Use create-react-app to create ReactJS boilerplate) says README.md was not auto-generated, yet there is no README.md in that commit.

(camh-anz)

> Add all generated source files except for `frontend/README.md`

(juliaorig commit)

I meant to say: I added all the source files that `create-react-app` generated except for the super long README.md that it generated - I deleted the `frontend/README.md` instead. To make it clearer I've aded to the commit messages:
> (Delete `frontend/README.md`).

> The third commit (Move Makefile to backend and update cloudbuid.yaml) also adds the steps for building the frontend, but the frontend Makefile is not added until the fifth commit.

Fixed (I believe).

> You modify the README.md in the fifth commit to document changes you made in an earlier commit (backend). Those doc changes ideally would go with the Makefile changes.

Yes I was aware of this before you commented on it. I thought I could slip it past you - in the end it wasn't actually to much effort to add it the relevant parts to the right commit, thanks for making me do this.
 
PTAL @camh-anz

juliaogris commented:
Ok, commit updated, PTAL @camh-anz
